scripts/package.lua

```diff
@@ -88,79 +98,85 @@
 	os.rmdir(pkgName)
 
 	print("Cloning source code")
-	z = os.executef("git clone .. %s", pkgName)
+	local z = execQuiet("git clone .. %s -b %s --recurse-submodules --depth 1 --shallow-submodules", pkgName, branch)
 	if not z then
 		error("clone failed", 0)
 	end
 
 	os.chdir(pkgName)
 
-	z = os.executef("git checkout %s", branch)
-	if not z then
-		error("unable to checkout branch " .. branch, 0)
-	end
+--
+-- Bootstrap Premake in the newly cloned repository
+--
 
-	z = os.executef("git submodule update --init")
+	print("Bootstraping Premake...")
+	if os.ishost("windows") then
+		z = execQuiet("Bootstrap.bat")
+	else
+		local os_map = {
+			linux = "linux",
+			macosx = "osx",
+		}
+		z = execQuiet("make -j -f Bootstrap.mak %s", os_map[os.host()])
```

Comment:
Should probably have a local for `os_map[os.host()]`, running this on BSD should fail more gracefully.

Additionally, perhaps the Bootstrap labels (Is that the correct terminology?) should match the Premake internal names? Having `os.host()` passed straight into _our_ Bootstrap.mak makes sense to me, thoughts? (Another PR for sure!)